strawberryfields/apps/qchem/dynamics.py

```diff
@@ -250,3 +254,81 @@ def prob(samples: list, excited_state: list) -> float:
         raise ValueError("The excited state must not contain negative values")
 
     return samples.count(excited_state) / len(samples)
+
+
+def sample_tmsv(
+    r: list, t: float, Ul: np.ndarray, w: np.ndarray, n_samples: int, loss: float = 0.0,
+) -> list:
+    r"""Generate samples for simulating vibrational quantum dynamics with an input two-mode squeezed state.
+
+    Given N two-mode squeezing parameters and N-dim normal-to-local transformation matrix, Ul,
+    a GBS device with 2N modes is simulated. The evolution operator acts on only the first N
+    modes in the device. A sample is generated by measuring the number of photons in each of
+    the :math:`2N` modes.
+
+    **Example usage:**
+
+    >>> r = [0.2, 0.8]
+    >>> t = 10.0
+    >>> Ul = np.array([[0.707106781, -0.707106781],
+    >>>                [0.707106781, 0.707106781]])
+    >>> w = np.array([3914.92, 3787.59])
+    >>> n_samples = 5
+    >>> sample_tmsv(r, t, Ul, w, n_samples)
+    [[0, 0, 0, 0], [0, 1, 0, 1], [0, 2, 0, 2], [0, 0, 0, 0], [1, 1, 1, 1]]
+
+    Args:
+        r (list): list of two-mode squeezing parameters
+        t (float): time in femtoseconds
+        Ul (array): normal-to-local transformation matrix
+        w (array): normal mode frequencies :math:`\omega` in units of :math:`\mbox{cm}^{-1}`
+        n_samples (int): number of samples to be generated
+        loss (float): loss parameter denoting the fraction of lost photons
+
+    Returns:
+        list[list[int]]: a list of samples
+    """
+    if n_samples < 1:
+        raise ValueError("Number of samples must be at least one")
+
+    if t < 0:
+        raise ValueError("Time must be zero or positive")
+
+    if np.any(w <= 0):
+        raise ValueError("Vibrational frequencies must be larger than zero")
+
+    if np.any(np.iscomplex(Ul)):
+        raise ValueError("The normal mode to local mode transformation matrix must be real")
+
+    if not len(r) == len(Ul):
+        raise ValueError(
+            "Number of squeezing parameters and the number of modes in the normal-to-local"
+            " transformation matrix must be equal"
+        )
+
+    N = len(Ul)
+    op = evolution(N)
+
+    eng = sf.LocalEngine(backend="gaussian")
+    prog = sf.Program(2 * N)
+
+    # pylint: disable=expression-not-assigned,pointless-statement
+    with prog.context as q:
+
+        for i in range(N):
+            sf.ops.S2gate(r[i]) | (q[i], q[i + N])
+
+        op(t, Ul, w) | q[:N]
+
+        if loss:
+            for _q in q:
+                sf.ops.LossChannel(1 - loss) | _q
+
+        sf.ops.MeasureFock() | q
+
+    with warnings.catch_warnings():
```

Comment:
What is the purpose of this?

Reply:
yeah I am not sure why this happens but we get a warning with the output when we run:
`xanadu/strawberryfields/strawberryfields/backends/gaussianbackend/backend.py:225: UserWarning: Cannot simulate non-Gaussian states. Conditional state after Fock measurement has not been updated.`
So Soran suppressed the warning. I do not know why it comes though.

Reply:
This is also in the standard `sample.sample()` function - the idea is to catch the warning that @AroosaIjaz shared. This warning happens when you have Fock measurements in the Gaussian backend, but I think it's unnecessary in the apps layer.